### ISSUE TITLE
Oversampling of under-populated samples

### ISSUE THREAD

**Issue by lov435** [lov435/SoftwareImprovement#12]
Possibly, our training data is not quite balanced well. Consider the following example"

Let's say a post has 10 comments which are divided into 5 groups of 2 comments each. This will genrate 10C2 = 45 pairs. Out of these 45 pairs only 5 are related (have the same group) and other 40 are not. So the model in this case will not have enough positive samples.

**Comment by lov435** [comment]
Hi Dr @damevski, A quick note. We have two classes for each pair of comments - related (1) and not-related (0). I printed out the number of pairs for each of the two classes. 

```
Size of Training set is 11766 11766
Y samples are  [(0, 6798), (1, 4968)]
```

So as you can see, our training data is not quite unbalanced as opposed to what we had thought.

**Comment by lov435** [comment]
I am using the package [imblearn](https://imbalanced-learn.readthedocs.io/en/stable/api.html) to implement SMOTE over sampling of the minority class samples. The version of this package I am using is 0.7.0. Please note that this version only works with `scikit-learn` version 0.23.1. Look at my SO [post](https://stackoverflow.com/a/63166278/5749570).

**Comment by lov435** [comment]
After increasing the minority class samples to match the number of majority class samples using SMOTE, the F-Measure seems to have gone up from 0.6925944638144215 to 0.7239848292461651 but the accuracy has gone down from 0.7716288384512684 to 0.7568347725403626. This makes sense because the training data as is are not that unbalanced. I am closing this issue with code checked in.
